### PR TITLE
New: Add tag ids & names to custom script variables

### PR DESCRIPTION
**Pull request by austinwbest**
#### Database Migration
NO

#### Description
New: Add tag ids & names to custom script variables

### CODE REVIEW

**Comment by ta264** [comment]
Having to use the repo to avoid a circular dependency is a bit of a warning sign for me.

Rather than fishing out the names of the tags and returning those I think you should just return the tag IDs (which you already have), and it's up to the user of the API to query the tag names if they care.

**Comment by austinwbest** [comment]
Ok, switched to tag ids then :) so a user can query the tag text from the API or map it in their custom scripts to the tag ids

**Comment by ta264** [comment]
I think that makes more sense to me but happy to hear arguments for the alternative

File: src/NzbDrone.Core/Notifications/CustomScript/CustomScript.cs [hunk 1]
```diff
@@ -48,6 +48,7 @@ public override void OnGrab(GrabMessage message)
             environmentVariables.Add("Radarr_Movie_TmdbId", movie.TmdbId.ToString());
             environmentVariables.Add("Radarr_Movie_In_Cinemas_Date", movie.InCinemas.ToString() ?? string.Empty);
             environmentVariables.Add("Radarr_Movie_Physical_Release_Date", movie.PhysicalRelease.ToString() ?? string.Empty);
+            environmentVariables.Add("Radarr_Movie_Tag_Ids", string.Join("|", movie.Tags.Select(t => t.ToString()).ToArray()));
```
Comment (ta264) [inline src/NzbDrone.Core/Notifications/CustomScript/CustomScript.cs:51]: Is the toarray needed? I think it'll work without.

**Comment by austinwbest** [comment]
Well my perspective to it differs from yours for sure. I believe the purpose of the variables is to avoid having end users do the leg work.. basically use them as is, otherwise might as well pass the movieId and let them API whatever they want. 

I totally understand why you prefer to not use the repo as even Q said yall had issues with that in the past, however in this edge case it wouldn't hurt and it doesn't require re-factoring to split the details out into their own files but does allow for returning the actual tag definition.

I would think repo usage would be case by case and if it where to create issues on the backend, then disallow it. Just my two cents, but as i told him yall are the owners and i'm just a guy on the outside helping out lol. So therefor i swapped it to your suggestion.

File: src/NzbDrone.Core/Notifications/CustomScript/CustomScript.cs (hunk 1 above)
Comment (austinwbest) [reply src/NzbDrone.Core/Notifications/CustomScript/CustomScript.cs]: You are correct, it is not needed. Oversight on my part on other tests and leaving it there. Ive removed it and re-pushed it.

**Comment by ta264** [comment]
That's a fair point, the rest of the environment variables are more friendly. Maybe it is better with the labels.

We were just guys on the outside helping out to start with too :)

**Comment by austinwbest** [comment]
So your blessing to put it back to the actual tag definition instead of id? I'm +1 for that! lol

File: src/NzbDrone.Core/Notifications/CustomScript/CustomScript.cs [hunk 2]
```diff
@@ -20,11 +21,15 @@ public class CustomScript : NotificationBase<CustomScriptSettings>
         private readonly IProcessProvider _processProvider;
         private readonly Logger _logger;
 
-        public CustomScript(IDiskProvider diskProvider, IProcessProvider processProvider, Logger logger)
+        // Have to use _tagRepo instead of _tagService because the service creates a crashing loop :|
+        private readonly ITagRepository _tagRepo;
```
Comment (Qstick) [inline src/NzbDrone.Core/Notifications/CustomScript/CustomScript.cs:25]: Separate TagService into TagService and TagDetailsService. Then you should be able to call TagService from here.

**Pull request closed without merging**